### ISSUE TITLE
"*** Can't get with persistent HTTP: Status code 302 while fetching"

### ISSUE THREAD

**Issue by SpencerDawkins** [martinthomson/i-d-template#400]
I'm not sure whether I should be reporting this in the i-d-template repo or in the kramdown-rfc repo, but it seemed that  i-d-template was a reasonable place to start. If I got that wrong, please let me know. 

In the past couple of weeks (and I wasn't doing much with GitHub for I-Ds for a few weeks prior, so I'm not sure when this might have started), my make command has started popping up these messages when renewing stale references: 

> /home/spencer/.cache/xml2rfc/reference.RFC.3550.xml: renewing (stale by 8.0 days) from https://www.rfc-editor.org/refs/bibxml/reference.RFC.3550.xml
> *** Can't get with persistent HTTP: Status code 302 while fetching https://www.rfc-editor.org/refs/bibxml/reference.RFC.3550.xml

The make command is still making the i-ds I expect it to make, so it's still working fine, modulo the warnings, but I remember that we started using net-http-persistent not too long ago. 

I expect there was a reason for that, so wanted to let people know that it's stopped working, at least for me. 

Thanks!

**Comment by martinthomson** [comment]
Well, I'm not sure where this belongs either :)  A little bit here perhaps; a little bit kramdown-rfc; a little bit local setup.

Firstly, this is a redirect, which kramdown-rfc doesn't (know how to) follow.  (The redirect is to bib.ietf.org, which you can see by following the link.)

You might get some movement by updating kramdown-rfc.  I expect that Carsten has already fixed the issue.  Try `make update-deps` for that.  (That assumes that you are using an installation of kramdown-rfc that this repo manages; you probably have a version installed on your system as well, which you can probably remove if you only ever use this repo for managing drafts.)

The reason you haven't seen builds fail is that kramdown-rfc is very good about using cached copies of references.  Unless you add a reference to a new RFC that it has never seen before, it will happily build successfully forever.

**Comment by SpencerDawkins** [comment]
@martinthomson - thank you for the clues! I did get a fair amount of movement from make update-deps, including but not limited to a new net-http-persistent, now 4.0.2, and I'm not getting warnings about redirects, so I expect I'm good. 

One thing I'm somewhat confused about, is that both the i-d-template repo and the kramdown-rfc repo have lists of dependencies, and I wasn't sure what to do with that - pick one? Install the set union? Something else? 

If you and @cabo are both in San Francisco, maybe we could talk about what makes sense to you both, and I could help with any cleanup that seems useful.

**Comment by cabo** [comment]
``` 
CommitDate: Thu Jun 8 16:20:57 2023 +0200
1.6.35: No longer use rfc-editor.org for RFC references
“`

So we are a little behind, I'd say.
Should the bib lookup follow redirects?  Probably.
On the other hand, we want this to be fast, and a redirect points out a problem (like the one fixed in 1.6.35), so this might cause some laziness on my side.

I'm not physically in San Francisco.

**Comment by martinthomson** [comment]
Yeah, I'm not seeing an action here, now that Carsten is aware of the situation.  Good to be aware, but no actions left.
